### PR TITLE
Fix docker build failures for arm64 and restrict ray serve to ~2.10.0

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: python/artexplainer.Dockerfile [hunk 1]
```diff
@@ -9,7 +9,8 @@ ARG POETRY_HOME=/opt/poetry
 ARG POETRY_VERSION=1.7.1
 
 # Required for building packages for arm64 arch
-RUN apt-get update && apt-get install -y --no-install-recommends python3-dev build-essential
+RUN apt-get update && apt-get install -y --no-install-recommends python3-dev build-essential pkg-config libhdf5-dev && \
```
Comment (spolti) [inline python/artexplainer.Dockerfile:12]: for arm we could pin the h5py version to 2.10 (IIRC) instead doing it here.
wdyt?

Comment (spolti) [reply python/artexplainer.Dockerfile:12]: A similar thing https://github.com/opendatahub-io/modelmesh-runtime-adapter/pull/55

Comment (sivanantha321) [reply python/artexplainer.Dockerfile:12]: We already doing some compiling for ARM64 and after reading the [comment here](https://github.com/h5py/h5py/pull/2405#issuecomment-2047248192) I am not sure when the ARM builds will be available.

Comment (spolti) [reply python/artexplainer.Dockerfile:12]: Oh, for now we could stick with 3.10 and wait for that. =/

Comment (spolti) [reply python/artexplainer.Dockerfile:12]: we still can use h5py with newer python versions, right?

**Pull request closed without merging**